### ISSUE TITLE
Question: Run env setup args before cmake commands?

### ISSUE THREAD

**Issue by rohit-kumar-j** [Civitasv/cmake-tools.nvim#137]
There are so man new features now!😄  

I was wondering if there was a way to send a command to the terminal or the quickfix list job to run so as to setup environment variables before executing a script?

Something like, and this would be taken from the `CMakeKits.json` :

```bash
~$ \home\script.sh
~$ cmake ..  #other cmake commands...
```

**Comment by rohit-kumar-j** [comment]
I think this is more relevant for the MSVC compiler as ninja works only when the `vcvars64.bat` file is executed. This sets up x64 native tools in the current shell. ~(Why Windows? Why can't you just work? Why do you hate me so much?)~ 

Something like:
```bash
~$ & "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvars64.bat"
**********************************************************************
** Visual Studio 2022 Developer Command Prompt v17.7.1
** Copyright (c) 2022 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
~$
~$
~$ cmake .. -GNinja
```
(The `&` at the beginning is important in windows)

**Comment by rohit-kumar-j** [comment]
This is trivial in terminal implementation, but I think that it needs to be run each time for the quickfix list implementation.

**Comment by Civitasv** [comment]
Why won’t it work if you have added ninja to the path variable?

**Comment by rohit-kumar-j** [comment]
> Why won’t it work if you have added ninja to the path variable?

Not exactly... it detects ninja. (with `scoop install ninja`), but apparently, the `vcvars64.bat` file needs to be run to setup other auxiliary tooling so that Clang and MSVC can work with ninja.

**Comment by rohit-kumar-j** [comment]
I think this can be easily run in the terminal once by passing the command at the very beginning, but I do not think that the same is possible for quickfix lists where it may need to be passed in each time, or have a bash profile that launches the shell with the setup script. Is this feature desirable on linux?

I'm sure people generally do not use msvc on linux, but are there any circumstances where some script needs to be run before cmake commands?

**Comment by Civitasv** [comment]
I think we only need run this script once. Will it work if you run this script before you run nvim?

**Comment by rohit-kumar-j** [comment]
> I think we only need run this script once. Will it work if you run this script before you run nvim?

I don't think so. Unless we ask vim to explicitly run it as a argument to the deafult shell. Currently the shell command to Powershell is `pwsh -NoLogo` (this is Powershell Core, not Windows Powershell). Either the user needs to specify this in their `*.ps1` (like how we use neofetch in the `.bashrc`) or as an explicit argument to vim. Maybe something like: 
```bash
pwsh -NoLogo -Argument  " & \"C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Auxiliary\\Build\\vcvars64.bat\" " 
```

**Comment by Civitasv** [comment]
> Maybe something like:
> 
> ```shell
> pwsh -NoLogo -Argument  " & \"C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Auxiliary\\Build\\vcvars64.bat\" " 
> ```

This might a solution to your problem.

To my understand, it should not be cmake-tools's responsiblity to set cmake environment.

**Comment by rohit-kumar-j** [comment]
> This might a solution to your problem.
> 
> To my understand, it should not be cmake-tools's responsiblity to set cmake environment.

Yes, I agree. However, I was under the impression that env scripts were intended to be implemented in the future. 

Ref: [TODO](https://github.com/Civitasv/cmake-tools.nvim/blob/master/docs/cmake_kits.md#todo)

**Comment by Civitasv** [comment]
> However, I was under the impression that env scripts were intended to be implemented in the future.

Yes, that is a nice feature. I will think how to implement it these days.
